lib/reek/context/code_context.rb

```diff
@@ -144,16 +143,35 @@ def config_for(detector_class)
           configuration_via_code_commment[detector_class.smell_type] || {})
       end
 
-      def track_visibility(visibility, names)
-        visibility_tracker.track_visibility children: children,
-                                            visibility: visibility,
-                                            names: names
-      end
-
       def number_of_statements
         statement_counter.value
       end
 
+      def module_context?
+        false
+      end
+
+      def method_context?
+        false
+      end
+
+      def singleton_method?
+        false
+      end
+
+      def instance_method?
+        false
+      end
+
+      def defined_as_instance_method?
+        false
```

Comment:
And additionally I don't think we should do it like that but rather use a map, please see my comment [here](https://github.com/troessner/reek/pull/796/files#diff-e4ff3ae7e7f4c263b7ab67ca5bb23e1fR20).